internal/rdb/rdb.go

```diff
@@ -134,6 +145,23 @@ func (r *RDB) Enqueue(ctx context.Context, msg *base.TaskMessage) error {
 	return nil
 }
 
+func (r *RDB) runIfNeeded(queue string, fn func() error) error {
```

Comment:
This approach uses a simple mutex instead of RW mutex. Contention shouldn't be an issue since most of the time will be spent issuing Redis network calls to enqueue the task payload. This approach also ensures only a single network call is sent to add the queue to "asynq:queues" set for better optimization.